AnimationPlayground.playground/Contents.swift

```diff
@@ -0,0 +1,437 @@
+import UIKit
+import XCPlayground
+
+let liveViewWidth :CGFloat = 400.0
+let liveView = UIView(frame: CGRect(x: 0.0, y: 0.0, width: liveViewWidth, height: 2000.0))
+liveView.backgroundColor = UIColor.whiteColor()
+XCPlaygroundPage.currentPage.liveView = liveView
+
+
+
+// Express position/widths etc in relation to the view's width (i.e. "lineWidth = w * 0.02" instead of "lineWidth = 3") so all drawing scales proportionally with the animationView's size
+
+
+
+struct WMF_Welcome_Plus {
+    var position: CGPoint
+    var opacity: Double
+    var initialTransform: CATransform3D
+    var finalTransform: CATransform3D
+}
+
+struct WMF_Welcome_Circle {
+    var position: CGPoint
+    var radius: CGFloat
+    var opacity: Double
+    var isDashed: Bool
+    var initialTransform: CATransform3D
+    var finalTransform: CATransform3D
+}
+
+struct WMF_Welcome_Line {
+    var position: CGPoint
+    var width: CGFloat
+    var opacity: Double
+    var initialTransform: CATransform3D
+    var finalTransform: CATransform3D
+}
+
+struct WMF_Welcome_Image {
+    var name: String
+    var frame: CGRect
+    var zPosition: CGFloat
+    var initialTransform: CGAffineTransform
+    var finalTransform: CGAffineTransform
+}
+
+struct WMF_Welcome_Bar {
+    var frame: CGRect
+    var color: UIColor
+    var cornerRadius: CGFloat
+    var opacity: Double
+    var initialTransform: CATransform3D
+    var finalTransform: CATransform3D
+}
+
+extension UIBezierPath {
+    class func wmf_circlePathWithRadius(radius: CGFloat) -> UIBezierPath {
+        return UIBezierPath(
+            arcCenter: CGPointZero,
+            radius: radius,
+            startAngle: 0.0,
+            endAngle: CGFloat(M_PI * 2.0),
+            clockwise: true
+        )
+    }
+    class func wmf_horizontalLinePathWithWidth(width: CGFloat) -> UIBezierPath {
+        let path = UIBezierPath()
+        path.moveToPoint(CGPointMake(width * -0.5, 0.0))
+        path.addLineToPoint(CGPointMake(width * 0.5, 0.0))
+        return path
+    }
+    class func wmf_verticalLinePathWithHeight(height: CGFloat) -> UIBezierPath {
+        let path = UIBezierPath()
+        path.moveToPoint(CGPointMake(0.0, height * -0.5))
+        path.addLineToPoint(CGPointMake(0.0, height * 0.5))
+        return path
+    }
+    class func wmf_plusPathWithWidth(width: CGFloat) -> UIBezierPath {
+        let path = UIBezierPath.wmf_horizontalLinePathWithWidth(width)
+        path.appendPath(UIBezierPath.wmf_verticalLinePathWithHeight(width))
+        return path
+    }
+}
+
+extension CAShapeLayer {
+    func wmf_applyCommonSettingsForSize(size: CGSize, position: CGPoint) -> Void {
+        self.lineCap = "round"
+        self.opacity = 0.0
+        self.lineWidth = size.width * 0.014
+        self.strokeEnd = 1.0
+        self.position = CGPointMake(size.width * position.x, size.height * position.y)
+        self.zPosition = 100
+    }
+    class func wmf_addLine(line: WMF_Welcome_Line, toView: UIView) -> CAShapeLayer {
```

Comment:
why is this a static function on `CAShapeLayer`? IMO would make more sense to extend `UIView` to `addWelcomeLine(line: WelcomeLine)`